bob-backend/src/pearl/core.rs

```diff
@@ -242,14 +242,36 @@ impl BackendStorage for Pearl {
         }
 
         if let Some(h) = oldest {
-            let memory = h.index_memory().await;
+            let memory = h.active_index_memory().await;
             h.close_active_blob().await;
             Some(memory)
         } else {
             None
         }
     }
 
+    async fn free_least_used_resources(&self) -> Option<usize> {
+        let mut least_modified: Option<Holder> = None;
+        let mut min_modification = u64::MAX;
+        for dc in self.disk_controllers.iter() {
+            if let Some(holder) = dc.find_least_modified_freeable_holder().await {
+                let modification = holder.last_modification().await;
+                if modification < min_modification {
+                    least_modified = Some(holder);
+                    min_modification = modification;
+                }
+            }
+        }
+
+        if let Some(h) = least_modified {
+            let memory = h.freeable_resources_memory().await;
+            h.free_freeable_resources().await;
```

Comment:
This function can return the amount of freed memory. In this way the separate function `freeable_resources_memory` is not needed